src/Main.elm

```diff
@@ -1,6 +1,8 @@
 module Main exposing (..)
 
 import Browser
+import Element
+import Element.Input
```

Comment:
okay, so these imports correspond to the UI library that I installed. If I hadn't installed them, this would not work. And like any library, it provides a namespace of functions (and elements?) that I can use in my code.

Reply:
Exactly!

Reply:
Have a look at the Elm UI package documentation for examples:

- https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/
- https://package.elm-lang.org/packages/mdgriffith/elm-ui/latest/Element

Reply:
I'm going to clarify something. You asked: 

> And like any library, it provides a namespace of functions (and elements?) that I can use in my code.

There are no 'elements' in Elm. The `Element` above exists solely because @mdgriffith decided to name the top level module in the Elm UI package `Element` since it was based on his work in [mdgriffith/style-elements](https://github.com/mdgriffith/style-elements).

Elm has *functions* and *values*: https://guide.elm-lang.org/core_language.html

> The smallest building block in Elm is called a *value*. This includes things like `42`, `True`, and `"Hello!"`.

> A *function* is a way to transform values. Take in one value, and produce another.

Values as well as functions have *types*: https://guide.elm-lang.org/types/reading_types.html

Most of the types you'll come across are:

- Primitives (String, Int, Bool, Float)
- Lists (ordered collection of values)
- Records (a bit like a type-safe dictionary - e.g. a value like: `{ firstName = "Hermann", lastName = "Hesse" }` has type `{ firstName: String, lastName: String}`)
- Custom types (like the `Msg` type we define on line 29)

Elm libraries (such as Elm UI and Html) expose *functions* and *type declarations*.

That's about it. In Elm you solve problems by thinking of data flowing through functions which take values of some type and return values of some other types. 

You might think of it like the Unix pipeline with strict type-checking added.

I hope this helps.